File: backend/src/main/java/net/zorphy/backend/main/dto/game/stats/GameStats.java
package net.zorphy.backend.main.dto.game.stats;

import net.zorphy.backend.main.dto.player.PlayerDetails;

import java.time.Duration;
import java.util.List;


public record GameStats(
        PlayerDetails player,
        int gamesPlayed,
        double winRate,
        LinkedGameStats<Integer> minScore,
        LinkedGameStats<Integer> maxScore,
        double avgScore,
        LinkedGameStats<Duration> minDuration,
        LinkedGameStats<Duration> maxDuration,
        Duration avgDuration,
        PlayerDetails nemesis,
        PlayerDetails victim,
        PlayerDetails rival,
        PlayerDetails companion,
        List<CorrelationResult> correlations,
        GameSpecificStats gameSpecific
) {
}

File: frontend/src/app/main/dto/games/stats/GameStats.ts
import {PlayerDetails} from "../../all/PlayerDetails";
import {ChartData} from "./ChartData";
import {CorrelationResult} from "./CorrelationResult";
import {LinkedGameStats} from "./LinkedGameStats";

export interface GameStats {
    player: PlayerDetails;
    gamesPlayed: number;
    winRate: number;
    minScore: LinkedGameStats<number>;
    avgScore: number;
    maxScore: LinkedGameStats<number>;
    minDuration: LinkedGameStats<string>;
    avgDuration: string;
    maxDuration: LinkedGameStats<string>;
    nemesis: PlayerDetails | null;
    victim: PlayerDetails | null;
    rival: PlayerDetails | null;
    companion: PlayerDetails | null;
    correlations: CorrelationResult[];
    gameSpecific: any;
}